### PR TITLE
Add registry bitness to RegUtil and BUtil.

### PR DESCRIPTION
**Pull request by barnson**
Fixes https://github.com/wixtoolset/issues/issues/6669.
Fixes https://github.com/wixtoolset/issues/issues/6670.

### CODE REVIEW

File: src/libs/dutil/WixToolset.DUtil/butil.cpp [hunk 1]
```diff
@@ -153,113 +162,26 @@ DAPI_(HRESULT) BundleEnumRelatedBundle(
...
-        ReleaseRegKey(hkBundle);
-        ReleaseNullStr(sczUninstallSubKey);
-        ReleaseNullStr(sczUninstallSubKeyPath);
+#if defined(_WIN64)
+        hr = DoBundleEnumRelatedBundle(hkRoot, REG_KEY_32BIT, wzUpgradeCode, pdwStartIndex, psczBundleId);
+        ButilExitOnFailure(hr, "Failed to enumerate 32-bit bundles.");
+#else
+        hr = DoBundleEnumRelatedBundle(hkRoot, REG_KEY_64BIT, wzUpgradeCode, pdwStartIndex, psczBundleId);
+        ButilExitOnFailure(hr, "Failed to enumerate 64-bit bundles.");
+#endif
```
Comment (rseanhall) [inline src/libs/dutil/WixToolset.DUtil/butil.cpp:181]: Why isn't Burn doing this? Don't 32-bit bundles need to detect 64-bit related bundles?

File: src/libs/dutil/WixToolset.DUtil/butil.cpp [hunk 2]
```diff
@@ -380,8 +302,9 @@ static HRESULT LocateAndQueryBundleValue(
 
     *pStatus = INTERNAL_BUNDLE_STATUS_SUCCESS;
 
-    if (FAILED(hr = OpenBundleKey(wzBundleId, BUNDLE_INSTALL_CONTEXT_MACHINE, wzSubKey, phKey)) &&
-        FAILED(hr = OpenBundleKey(wzBundleId, BUNDLE_INSTALL_CONTEXT_USER, wzSubKey, phKey)))
+    if (FAILED(hr = OpenBundleKey(wzBundleId, BUNDLE_INSTALL_CONTEXT_MACHINE, wzSubKey, REG_KEY_32BIT, phKey)) &&
+        FAILED(hr = OpenBundleKey(wzBundleId, BUNDLE_INSTALL_CONTEXT_MACHINE, wzSubKey, REG_KEY_64BIT, phKey)) &&
+        FAILED(hr = OpenBundleKey(wzBundleId, BUNDLE_INSTALL_CONTEXT_USER, wzSubKey, REG_KEY_DEFAULT, phKey)))
```
Comment (rseanhall) [inline src/libs/dutil/WixToolset.DUtil/butil.cpp:307]: Again, this seems to be different from Burn.

File: src/libs/dutil/WixToolset.DUtil/butil.cpp [hunk 3]
```diff
@@ -472,3 +396,116 @@ static HRESULT CopyStringToBuffer(
...
     return hr;
 }
+
+static HRESULT DoBundleEnumRelatedBundle(
+    __in HKEY hkRoot,
+    __in REG_KEY_BITNESS kbKeyBitness,
+    __in_z LPCWSTR wzUpgradeCode,
+    __inout PDWORD pdwStartIndex,
+    __deref_out_z LPWSTR* psczBundleId
+)
```
Comment (rseanhall) [inline src/libs/dutil/WixToolset.DUtil/butil.cpp:406]: We need to get the .editorconfig working for C++.

File: src/libs/dutil/WixToolset.DUtil/regutil.cpp [hunk 4]
```diff
@@ -936,6 +940,25 @@ DAPI_(BOOL) RegValueExists(
...
 
+static REGSAM TranslateKeyBitness(
+    __in REG_KEY_BITNESS kbKeyBitness
+    )
+{
+    switch (kbKeyBitness)
+    {
+    case REG_KEY_32BIT:
+        return KEY_WOW64_32KEY;
+        break;
```
Comment (rseanhall) [inline src/libs/dutil/WixToolset.DUtil/regutil.cpp:951]: The `break`s are unreachable due to the `return`s.

File: src/libs/dutil/WixToolset.DUtil/butil.cpp [hunk 5]
```diff
@@ -153,113 +162,26 @@ DAPI_(HRESULT) BundleEnumRelatedBundle(
...
-
-            break;
-        }
-
-        // Cleanup before next iteration
-        ReleaseRegKey(hkBundle);
-        ReleaseNullStr(sczUninstallSubKey);
-        ReleaseNullStr(sczUninstallSubKeyPath);
+#if defined(_WIN64)
+        hr = DoBundleEnumRelatedBundle(hkRoot, REG_KEY_32BIT, wzUpgradeCode, pdwStartIndex, psczBundleId);
```
Comment (rseanhall) [inline src/libs/dutil/WixToolset.DUtil/butil.cpp:176]: `pdwStartIndex` was a shortcut for a context parameter but now we have to keep track of the current bitness, too. This function will skip the first `*pdwStartIndex` entries here.

**Comment by barnson** [comment]
Yep, making another pass on Burn.

**Pull request merged**